lib/livebook/utils.ex

```diff
@@ -112,17 +112,17 @@ defmodule Livebook.Utils do
       ** (RuntimeError) Livebook.Utils.access_by_id/1 expected a list, got: %{}
 
   """
-  @spec access_by_id(term()) ::
+  @spec access_by_id(term(), atom()) ::
           Access.access_fun(data :: struct() | map(), current_value :: term())
-  def access_by_id(id) do
+  def access_by_id(id, field \\ :id) do
     fn
       :get, data, next when is_list(data) ->
         data
-        |> Enum.find(fn item -> item.id == id end)
+        |> Enum.find(fn item -> Map.get(item, field) == id end)
```

Comment:
is there a reason why item[field] is not working?

Reply:
In other places we use this with structs, like `%Cell{}`, and `item[field]` invokes the `Access` protocol, which is not implemented on these structs.

Here what we want is `Map.fetch!(item, field)`, since we always expect the field to be there.